RecoEgamma/ElectronIdentification/plugins/ElectronMVAEstimatorRun2Phys14NonTrig.cc

```diff
@@ -213,85 +219,130 @@ void ElectronMVAEstimatorRun2Phys14NonTrig::fillMVAVariables(const edm::Ptr<reco
-  _allMVAVars.dphi            = eleRecoPtr->deltaPhiSuperClusterTrackAtVtx();
-  _allMVAVars.detacalo        = eleRecoPtr->deltaEtaSeedClusterTrackAtCalo();
+  allMVAVars.deta            = eleRecoPtr->deltaEtaSuperClusterTrackAtVtx();
+  allMVAVars.dphi            = eleRecoPtr->deltaPhiSuperClusterTrackAtVtx();
+  allMVAVars.detacalo        = eleRecoPtr->deltaEtaSeedClusterTrackAtCalo();
   // Spectator variables  
-  _allMVAVars.pt              = eleRecoPtr->pt();
-  float scEta = superCluster->eta();
-  _allMVAVars.isBarrel        = ( std::abs(scEta) < 1.479 );
-  _allMVAVars.isEndcap        = ( std::abs(scEta) >= 1.479);
-  _allMVAVars.SCeta           = scEta;
-
+  allMVAVars.pt              = eleRecoPtr->pt();
+  const float scEta = superCluster->eta();
+  allMVAVars.isBarrel        = ( std::abs(scEta) < 1.479 );
+  allMVAVars.isEndcap        = ( std::abs(scEta) >= 1.479);
+  allMVAVars.SCeta           = scEta;
+
+  constrainMVAVariables(allMVAVars);
+
+  std::vector<float> vars;
+
+  if( isEndcapCategory( findCategory(particle) ) ) {
+    vars = std::move( packMVAVariables( allMVAVars.kfhits,
+                                        allMVAVars.see,
+                                        allMVAVars.spp,
+                                        allMVAVars.OneMinusE1x5E5x5,
+                                        allMVAVars.R9,
+                                        allMVAVars.etawidth,
+                                        allMVAVars.phiwidth,
+                                        allMVAVars.HoE,
+                                        allMVAVars.PreShowerOverRaw,
+                                        allMVAVars.kfchi2,
+                                        allMVAVars.gsfchi2,
+                                        allMVAVars.fbrem,
+                                        allMVAVars.EoP,
+                                        allMVAVars.eleEoPout,
+                                        allMVAVars.IoEmIoP,
+                                        allMVAVars.deta,
+                                        allMVAVars.dphi,
+                                        allMVAVars.detacalo,
+                                        allMVAVars.pt,
+                                        allMVAVars.isBarrel,
+                                        allMVAVars.isEndcap,
+                                        allMVAVars.SCeta )
+                      );
+  } else {
+    vars = std::move( packMVAVariables( allMVAVars.kfhits,
+                                        allMVAVars.see,
+                                        allMVAVars.spp,
+                                        allMVAVars.OneMinusE1x5E5x5,
+                                        allMVAVars.R9,
+                                        allMVAVars.etawidth,
+                                        allMVAVars.phiwidth,
+                                        allMVAVars.HoE,
+                                        allMVAVars.kfchi2,
```

Comment:
this ~20-line copy-paste with only one line change is yet another poorly maintainable piece.